pkg/util.go

```diff
@@ -271,3 +273,54 @@ var (
 		time.TimeOnly,
 	}
 )
+
+func ExtractHostAndPort(curlCmd string) (string, string, error) {
+	// Split the command string to find the URL
+	parts := strings.Split(curlCmd, " ")
+	for _, part := range parts {
+		if strings.HasPrefix(part, "http") {
+			u, err := url.Parse(part)
+			if err != nil {
+				return "", "", err
+			}
+			host := u.Hostname()
+			port := u.Port()
+			if port == "" {
+				if u.Scheme == "https" {
+					port = "443"
+				} else {
+					port = "80"
+				}
+			}
+			return host, port, nil
+		}
+	}
+	return "", "", fmt.Errorf("no URL found in CURL command")
+}
+
+func WaitForPort(ctx context.Context, host string, port string, timeout time.Duration) error {
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), 1*time.Second)
```

Comment:
since your ticker is 1 second, your timeout here should be less than 1 sec

Reply:
Didn't get this point. why should it be less than 1 sec?